### ISSUE TITLE
Deploying on the cluster a custom controller for an existing resource (ReplicaSet)

### ISSUE THREAD

**Issue by Laurianti** [kubernetes-sigs/kubebuilder#1740]
Hello, I created a controller for an existing resource (ReplicaSet), implementing my own logic.
With "make run" it works perfectly.

I followed this steps to run it on the cluster:
make docker-build docker-push IMG=localhost.localdomain:5000/test-1:tag
make deploy IMG=localhost.localdomain:5000/test-1:tag

And all works perfectly.

But what I wrote inside my controller doesn't seem to happen.
Am I wrong something?

**Comment by camilamacedo86** [comment]
Hi @Laurianti, 

You did not provide detailed info to help you with. However, are you applying the CR(samples) for your API? You need to apply in the cluster the CR's to trigger the reconcile with `kubectl apply -f config/samples/`. More info: https://book.kubebuilder.io/quick-start.html

Are you trying to trigger the reconcile for an existent resource that is not created by the controller? If yes, you would need to implement it the watch feature : 

Note that one of the attributes that can be informed is `IsController` to the watches: 


```go

      err := c.Watch(&source.Kind{Type: obj}, &handler.EnqueueRequestForOwner{
		IsController: isConttroller,
		OwnerType:    owner,
	})

```

More info: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/handler#EnqueueRequestForOwner

I have a personal project with this example, see https://github.com/dev4devs-com/postgresql-operator/blob/master/pkg/controller/backup/controller.go#L60-L63 where I am watching a resource that is managed by another controller. See [here](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.6.3/pkg/builder#Builder.Watches) that the builder also allows set the Watches. Did you try to solve it in this way?

**Comment by FedericoLaurianti-ScaiConsulting** [comment]
Hi @camilamacedo86,

I haven't created a (CR)Custom Resource.

I executed this command:

kubebuilder create api --group apps --kind ReplicaSet --version v1

I pressed "n" for Create Resource [y/n] and "y" for Create Controller [y/n]

I want only a controller for the existing "ReplicaSet" resource.

And I want to deploy it on the kubernetes cluster.

What are the steps I have to do only this?

Thank you

**Comment by Laurianti** [comment]
Hi @camilamacedo86,

I haven't created a (CR)Custom Resource.

I executed this command:

kubebuilder create api --group apps --kind ReplicaSet --version v1

I pressed "n" for Create Resource [y/n] and "y" for Create Controller [y/n]

I want only a controller for the existing "ReplicaSet" resource.

And I want to deploy it on the kubernetes cluster.

What are the steps I have to do only this?

Thank you
